tomviz/DataSource.cxx

```diff
@@ -232,16 +232,26 @@ bool DataSource::appendSlice(vtkImageData* slice)
 
 void DataSource::setFileName(const QString& filename)
 {
-  auto reader = m_json.value("reader").toObject(QJsonObject());
-  reader["fileName"] = filename;
-  m_json["reader"] = reader;
+  // AG: remove fileName and fileNames redundancy. but keep backward
+  // compatibility.
+  QStringList fileNames = QStringList(filename);
+  setFileNames(fileNames);
+
+  // auto reader = m_json.value("reader").toObject(QJsonObject());
+  // reader["fileName"] = filename;
+  // m_json["reader"] = reader;
 }
 
 QString DataSource::fileName() const
 {
+  // AG: remove fileName and fileNames reduntancy. but keep backward
```

Comment:
As above, change looks reasonable but commit message would be better place to explain the change.